docs/README

```diff
@@ -1,23 +0,0 @@
-Welcome to Cython's documentation.
```

Comment:
Why do you think that this file is no longer useful?

Reply:
When I remove it, I have a plan to replace it with `README.rst` that has a new guide for newcomers to edit and build docs locally and link it to docs. But I realize the file have historical value to the project and is already `CONTRIBUTING.rst` in place for contributing guide